benchclients/python/README.md

```diff
@@ -1,3 +1,104 @@
 # benchclients
 
 A lightweight Python package with clients for working with APIs in the Conbench ecosystem
```

Comment:
Within the first few lines I'd love to understand whether this is a library or a CLI or both.

>  for working with APIs in the Conbench ecosystem

This sounds like we have various APIs to work with. I think there is just one, but maybe I don't know something here. Let's add precision :-).

Reply:
> Within the first few lines I'd love to understand whether this is a library or a CLI or both. 

My mental model is that "package" implies "library" unless CLI is mentioned, but yeah, we can switch to "library" if it's clearer.

> I do not care so much about the "lightweight".

Some users do; part of why this package exists is because the legacy runner required installing all the webapp deps, which not everybody wants to install in their CI.

> > for working with APIs in the Conbench ecosystem
> 
> This sounds like we have various APIs to work with. I think there is just one, but maybe I don't know something here. Let's add precision :-).

Yeah, there's a little ambiguity here because primarily this library addresses the one right now, but is designed to work with more if/when they exist; `benchalerts` extends the base client here to make a `GitHubClient`. If that becomes a thing others need, it should live here. So there's a difference between "what is currently here" and "what belongs here". I don't want to make this too verbose, but I'll see if I can clarify.

> If we can narrow down "working with the API" a tiny little further, I'd also love to see that.
> 
> Examples: is this supposed to support the _complete_ Conbench API surface? Or does it have the goal to help with submitting benchmark data?

It's very general and doesn't get specific to what endpoints exist, it just handles auth and retries and logging and such. So it's a slightly higher level than `requests` (though not much), but definitely isn't a mapping of every endpoint. Users will still probably need to reference the Swagger or another docs source to actually make useful requests with the client.

Reply:
>  we can switch to "library" if it's clearer.

Yes, please! 

> part of why this package exists is because the legacy runner required installing all the webapp deps, which not everybody wants to install in their CI.

That's a great point to know to understand the history of this library. 

But for new people coming to this ecosystem they don't know the legacy runner and "lightweight" means something different depending on the person who reads it.

> addresses the one right now, but is designed to work with more if/when they exist;

It's this level of ambiguity which makes it hard for new users and developers to understand what we build, and what the individual components do. One can describe the state of today with precision, without imposing road blocks for the future :).